lib/src/main/java/com/cloudbees/groovy/cps/impl/CpsCallableInvocation.java

```diff
@@ -64,6 +69,19 @@ void checkMismatch(Object expectedReceiver, List<String> expectedMethodNames) {
             return; 
         }
 
+        if (expectedReceiver instanceof GroovyObject) {
+            Object propVal = ((GroovyObject) expectedReceiver).getMetaClass().getProperty(expectedReceiver, expectedMethodName);
```

Comment:
I thought about something like this, but what if the implementations of `MetaClass.getProperty` or `Map.get` have side effects? I guess we just assume that in most cases they won't, so this is safe?

Reply:
Not that we'd care here, I don't think. `MetaClass.getProperty` is definitely clean, `Map.get`...well, if it's not clean, then there are a bunch of other things that can go wrong outside of this scope. =)

Reply:
This isn't perfect, but it does deal with these scenarios for mismatch purposes.

Reply:
Fair enough. It just reminds me a lot of SECURITY-1465 (https://github.com/jenkinsci/groovy-sandbox/commit/e30cd28d7b30cd606e78c22174cb04e0450244a7), where we ran into issues calling user-defined methods in places that expected "normal" behavior. I don't think it would be a security issue here since the actual method would still be sandbox-transformed, it would just cause confusing behavior.